tools/devops/automation/templates/packages/build.yml

```
@@ -0,0 +1,454 @@
+    APP_ID: $(XamarinAppId)
+    INSTALL_ID: $(XamarinAppId)
+    APPLE_ACCOUNT: $(XamarinUserId)
+    APPLE_PASS: $(XamarinPassword)
+    TEAM_ID: $(TeamID)
+    KEYCHAIN: $(SigningKeychain)
+  name: notarize
+  displayName: 'Signing Release Build'
+  condition: and(succeeded(), eq(variables['configuration.SignPkgs'], 'True'), eq(variables['configuration.IsPr'], 'False'))
+  timeoutInMinutes: 90
+
+- template: generate-workspace-info.yml@templates
+  parameters:
+    GitHubToken: $(GitHub.Token)
+    ArtifactDirectory: $(Build.SourcesDirectory)/package-internal
+
+- template: uninstall-certificates/v1.yml@templates
+  parameters:
+    HostedMacKeychainPassword: $(OSX_KEYCHAIN_PASS)
+
+# upload each of the pkgs into the pipeline artifacts
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish Build Artifacts'
+  inputs:
+    targetPath: $(Build.SourcesDirectory)/package
+    artifactName: package
+  continueOnError: true
+
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish Build Internal Artifacts'
+  inputs:
+    targetPath: $(Build.SourcesDirectory)/package-internal 
+    artifactName: package-internal
+  continueOnError: true
+
+- bash: |
+    set -x
+    set -e
+
+    make -C $(Build.SourcesDirectory)/xamarin-macios/tests package-tests
+  displayName: 'Package Xamarin.mac tests'
+  condition: and(succeeded(), eq(variables['configuration.RunMacTests'], 'True'))
+  continueOnError: true # not a terrible blocking issue
+
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish Xamarin.Mac tests'
+  inputs:
+    targetPath: $(Build.SourcesDirectory)/xamarin-macios/tests/*.7z
+    artifactName: package-internal
+  condition: and(succeeded(), eq(variables['configuration.RunMacTests'], 'True'))
+  continueOnError: true
+
+- bash: |
+    make -j8 -C $(Build.SourcesDirectory)/xamarin-macios/tools/apidiff jenkins-api-diff
+
+    # remove some files that do not need to be uploaded
+    cd $(Build.SourcesDirectory)/xamarin-macios/tools/apidiff/
+    rm -Rf *.exe *.pdb *.stamp *.zip *.sh ./references ./temp
+  displayName: 'API diff (from stable)'
+  condition: and(succeeded(), eq(variables['configuration.SkipPublicJenkins'], 'False'))
+  continueOnError: true
+  env:
+    BUILD_REVISION: 'jenkins'
+
+- task: ArchiveFiles@1
+  displayName: 'Archive API diff (from stable)'
+  inputs:
+    rootFolder: $(Build.SourcesDirectory)/xamarin-macios/tools/apidiff
+    includeRootFolder: false
+    archiveFile: '$(Build.ArtifactStagingDirectory)/apidiff-stable.zip'
+  condition: and(succeeded(), eq(variables['configuration.SkipPublicJenkins'], 'False'))
+  continueOnError: true
+
+- task: PublishPipelineArtifact@1
+  displayName: 'Publish API diff (from stable)'
+  inputs:
+    targetPath: '$(Build.ArtifactStagingDirectory)/apidiff-stable.zip'
+    artifactName: apidiff-stable
+  condition: and(succeeded(), eq(variables['configuration.SkipPublicJenkins'], 'False'))
+  continueOnError: true
+
+- bash: |
+    echo "Running tests on $AGENT_NAME"
```

Make sure errors aren't ignored:

```suggestion
    set -x
    set -e
    echo "Running tests on $AGENT_NAME"
```